code/modules/maps/tether/lateload.dm

```diff
@@ -196,6 +206,24 @@
 /datum/map_z_level/tether_lateload/away_fueldepot
 	name = "Away Mission - Fuel Depot"
 
+// Class D (Virgo 5)
+
+/datum/map_template/tether_lateload/away_class_d
+	name = "Class D - Mountains and Rock Plains"
+	desc = "The previously nuked planet Class D away mission"
+	mappath = '_maps/map_levels/140x140/Class_D.dmm'
+	associated_map_datum = /datum/map_z_level/tether_lateload/away_class_d
+
+/datum/map_template/tether_lateload/away_class_d/on_map_loaded(z)
+	. = ..()
+	seed_submaps(list(z), 200, /area/poi_d/unexplored, /datum/map_template/submap/level_specific/class_d)
+	new /datum/random_map/automata/cave_system/no_cracks(null, 3, 3, z, world.maxx - 4, world.maxy - 4)
```

Comment:
You should remove line here it instructs the map to write over the existing shape of the world, which is something you do not want since it makes navigating Class D a pain...

Reply:
seeding minerals on a mineral world seems important though